webapp-django/crashstats/crashstats/tests/test_jinja_helpers.py

```diff
@@ -190,3 +193,93 @@ def test_basics(self):
         eq_(digitgroupseparator(1000), '1,000')
         eq_(digitgroupseparator(-1000), '-1,000')
         eq_(digitgroupseparator(1000000L), '1,000,000')
+
+
+class TestHumanizers(TestCase):
+
+    def test_show_duration(self):
+        html = show_duration(59)
+        ok_(isinstance(html, SafeText))
+        eq_(
+            html,
+            '59 seconds'
+        )
+
+        html = show_duration(150)
+        ok_(isinstance(html, SafeText))
+        eq_(
+            html,
+            '150 seconds <span class="humanized" title="150 seconds">'
+            '(2 minutes and 30 seconds)</span>'
+        )
+
+        # if the number is digit but a string it should work too
+        html = show_duration('150')
+        eq_(
+            html,
+            '150 seconds <span class="humanized" title="150 seconds">'
+            '(2 minutes and 30 seconds)</span>'
+        )
+
+    def test_show_duration_different_unit(self):
+        html = show_duration(150, unit='cool seconds')
+        ok_(isinstance(html, SafeText))
+        eq_(
+            html,
+            '150 cool seconds '
+            '<span class="humanized" title="150 cool seconds">'
+            '(2 minutes and 30 seconds)</span>'
+        )
+
+    def test_show_duration_failing(self):
+        html = show_duration(None)
+        eq_(html, None)
+        html = show_duration('not a number')
+        eq_(html, 'not a number')
```

Comment:
I would also add ``ok_(not isinstance(html, SafeText))`` to prove that those values are not marked safe.